assets/javascripts/discourse/controllers/docs-index.js

```diff
@@ -273,7 +273,7 @@ export default Controller.extend({
   updateSelectedTags(tag) {
     let filter = this.filterTags;
     if (filter && filter.includes(tag.id)) {
-      filter = filter.replace(tag.id, "").replace(/^\|+|\|+$/g, "");
```

Comment:
This regular expression tried to fix stuff like `foo||baz` after removing a tag, but because it is anchored to the start of the string it didn't work.